receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -40,6 +40,13 @@ type metricFamily struct {
 	labelKeysOrdered  []string
 	metadata          *scrape.MetricMetadata
 	groupOrders       map[string]int
+	exemplars         map[string][]exemplarData
```

Comment:
should we just directly construct a `pmetric.ExemplarSlice` here instead of storing an intermediary data type?